src/mono/mono/mini/aot-compiler.c

```diff
@@ -13055,15 +13053,16 @@ collect_methods (MonoAotCompile *acfg)
 		}
 		*/
 
-		if (should_emit_extra_method_for_generics (method, TRUE)) {
+		if (should_emit_extra_method_for_generics (method, TRUE, error)) {
 			/* Compile the ref shared version instead */
 			method = mini_get_shared_method_full (method, SHARE_MODE_NONE, error);
-			if (!method) {
-				aot_printerrf (acfg, "Failed to load method 0x%x from '%s' due to %s.\n", token, image->name, mono_error_get_message (error));
-				aot_printerrf (acfg, "Run with MONO_LOG_LEVEL=debug for more information.\n");
-				mono_error_cleanup (error);
-				return FALSE;
-			}
+		}
+
+		if (!is_ok (error) || !method) {
```

Comment:
I believe it is the case that we only ever get to this line with `method == NULL` if `should_emit_...` returns TRUE, but it's very hard to figure that out.

what if you had something like:
```c
gboolean warn_and_skip = FALSE;
if (should_emit (...)) {
   mono_error_assert_ok (error);
    method = mini_get_(...);
    
   if (!method || !is_ok (error)) {
       warn_and_skip = TRUE;
    }
} else if (!is_ok (error)) {
   warn_and_skip = TRUE;
}

if (warn_and_skip) {
    aot_printerrf(...);
}
```